### ISSUE TITLE
[0.32.1] Cannot push new hashes to aws, gitlab, gcp registries

### ISSUE THREAD

**Issue by xarses** [open-policy-agent/conftest#716]
I've been using `conftest push` in a script to push changes to our ecr oci repo after updating the policy codebase. After updating to (originally 0.32.0) 0.32.1 I've found that I'm now unable to upload new hashes. Uploading to a hash that already exists appears to work fine

```
conftest push oci://1234567890.dkr.ecr.us-east-1.amazonaws.com/terraform-policy:latest -p policy
2022/06/13 20:35:06 pushing bundle to: 1234567890.dkr.ecr.us-east-1.amazonaws.com/terraform-policy:latest
Error: push bundle: pushing manifest: failed commit on ref "manifest-sha256:83553ef1822eab5bc55398a6e5ecb4a2af65503fd187644679e627133c7c4862": unexpected status: 404 Not Found
```

Steps to reproduce:

* Change the contents of what your uploading
* upload to the ecr repo
* note the failure
* fall back to an old version
* upload (sucessfully)
* switch back to 0.32.1
* upload (upload appears happy)

(If you haven't already set up auth, you need to login via aws and docker cli first)
`aws ecr get-login-password --region us-east-1 | docker login --username AWS --password-stdin 1234567890.dkr.ecr.us-east-1.amazonaws.com`

Full upload script (sanitized)
```
# set the ECR repo path
export ECR="1234567890.dkr.ecr.us-east-1.amazonaws.com"

# ensure we have the current master
git checkout master
git pull 

# verify tests
conftest verify -d policy

# disable some BS in macos tar
export COPYFILE_DISABLE=1

# push to latest
conftest push oci://${ECR}/terraform-policy:latest

# a tag was given
if [ ! -z "$1" ]; then
    echo tagging for release
    git tag ${1}
    git push origin ${1}
    conftest push oci://${ECR}/terraform-policy:${1}
fi
```

**Comment by jalseth** [comment]
Hi @xarses , can you confirm the version of `conftest` you're using? There is no version 0.31.1. You can run `conftest --version` to confirm.

**Comment by xarses** [comment]
@jalseth my bad, version brain. the problem is visible on `0.32.0` and `0.32.1` its working as expected on 0.31.0

**Comment by tspearconquest** [comment]
I'm having the same issue on Gitlab Container Registry. I have been working with Gitlab Support to try to identify if it was conftest or Gitlab with the problem but the fact that this happens elsewhere confirms its conftest.

An update from Gitlab Support:

> I found where the digest that's not being found is being uploaded as a blob
> 
> Looking at the timestamp for the blob upload, we see that happens on Jun 21, 2022 @ 15:06:57.592 while the manifest push first fails on Jun 21, 2022 @ 15:02:54.374. From what I'm seeing, the client isn't properly pushing. It needs to wait until the layers are uploaded before pushing the manifest. This also explains why subsequent attempts are succeeding.

**Comment by jalseth** [comment]
@xarses @tspearconquest I've fixed this when pushing to GCP GAR via #721, can you test as well before I submit the change?

**Comment by srspnda-instacart** [comment]
> @xarses @tspearconquest I've fixed this when pushing to GCP GAR via #721, can you test as well before I submit the change?

Thanks @jalseth confirming for @xarses that this fixes our issue.

**Comment by jalseth** [comment]
Thanks for confirming! I'm thinking 2/3 successful tests is a good indicator that the change is successful so I'll merge it. I'll also be working on #718 to add e2e tests for this, so we don't run into the same issue again in the future.

@tspearconquest if you continue to have issues after the next release (I'll trigger that after merging this fix) please feel free to reopon.
